python/taulabs/telemetry.py

```diff
@@ -212,36 +268,139 @@ def __select(self, finishTime):
         if r:
             # Shouldn't throw an exception-- they just told us
             # it was ready for read.
-            chunk = self.sock.recv(1024)
+            chunk = os.read(self.fd, 1024)
             if chunk == '':
-                raise RuntimeError("socket closed")
+                raise RuntimeError("stream closed")
 
             self.recv_buf = self.recv_buf + chunk
 
             didStuff = True
 
         if w:
-            written = self.sock.send(self.send_buf)
-            self.send_buf = self.send_buf[written:]
+            written = os.write(self.fd, self.send_buf)
+
+            if written > 0:
+                self.send_buf = self.send_buf[written:]
 
             didStuff = True
 
         return didStuff
 
-    def __receive(self, finishTime):
-        """ Fetch available data from TCP socket """
+    def _send(self, msg):
+        """ Send a string out the TCP socket """
```

Comment:
This wouldn't have to be strictly a TCP socket now that you've abstracted out the FDTelemetry class.